src/Compilers/Core/Portable/SourceGeneration/GeneratorSyntaxWalker.cs

```diff
@@ -33,11 +32,14 @@ public void VisitWithModel(Lazy<SemanticModel>? model, SyntaxNode node)
             _semanticModel = null;
         }
 
-        public override void Visit(SyntaxNode node)
+        public void Visit(SyntaxNode node)
         {
-            Debug.Assert(_semanticModel is object && _semanticModel.Value.SyntaxTree == node.SyntaxTree);
-            _syntaxReceiver.OnVisitSyntaxNode(new GeneratorSyntaxContext(node, _semanticModel, _syntaxHelper));
-            base.Visit(node);
+            Debug.Assert(_semanticModel is object);
+            foreach (var child in node.DescendantNodesAndSelf())
+            {
+                Debug.Assert(_semanticModel.Value.SyntaxTree == child.SyntaxTree);
+                _syntaxReceiver.OnVisitSyntaxNode(new GeneratorSyntaxContext(child, _semanticModel, _syntaxHelper));
```

Comment:
>_syntaxReceiver.OnVisitSyntaxNode(new GeneratorSyntaxContext(child, _semanticModel, _syntaxHelper)); [](http://example.com/codeflow?start=16&length=100)

Is this going to perform callbacks in the same order? Is the order of call backs important?

Reply:
Yes, I believe it will be in the same order. DescendantNodesAndSelf and the previous usage of SyntaxWalker.Visit both do pre-order traversal